galleries/examples/images_contours_and_fields/image_exact_placement.py

```diff
@@ -0,0 +1,171 @@
+
+# plot with default axes handling:
+fig, axs = plt.subplots(1, 2, facecolor='aliceblue')
+
+axs[0].imshow(A, vmin=-1, vmax=1)
+axs[1].imshow(B, vmin=-1, vmax=1)
+
+
+def annotate_rect(ax):
+    # add a rectangle that is the size of the B matrix
+    rect = mpatches.Rectangle((0, 0), 200, 40, linewidth=1,
+                              edgecolor='r', facecolor='none')
+    ax.add_patch(rect)
+    return rect
+
+annotate_rect(axs[0])
+
+# %%
+# Note that both images are rendered at a 1:1 ratio, but are made to look almost the
+# same width, despite image B being smaller than image A.
+#
+# If the size of the images are amenable, we can preserve the relative sizes of two
+# images by using either the *width_ratio* or *height_ratio* of the subplots.  Which
+# one you use depends on the shape of the image and the size of the figure.
+# We can control the relative sizes using the *width_ratios* argument *if* the images
+# are wider than they are tall and shown side by side, as is the case here.
+#
+# While we are making changes, let us also make the aspect ratio of the figure closer
+# to the aspect ratio of the axes using *figsize* so that the figure does not have so
+# much white space.
+
+figsize = (6.4, 2)  # approximate figsize that trims some of the white space.
+
+fig, axs = plt.subplots(1, 2, width_ratios=[300/200, 1],
+                        figsize=figsize, facecolor='aliceblue')
+
+axs[0].imshow(A, vmin=-1, vmax=1)
+annotate_rect(axs[0])
+
+axs[1].imshow(B, vmin=-1, vmax=1)
+# %%
+# Given that the data subsample is in the upper left of the larger image,
+# it might make sense if the top of the smaller Axes aligned with the top of the larger.
+# This can be done manually by using `~.Axes.set_anchor`, and using "NW" (for
+# northwest).
+
+fig, axs = plt.subplots(1, 2, width_ratios=[300/200, 1],
+                        figsize=figsize, facecolor='aliceblue')
+
+axs[0].imshow(A, vmin=-1, vmax=1)
+annotate_rect(axs[0])
+
+axs[0].set_anchor('NW')
+axs[1].imshow(B, vmin=-1, vmax=1)
+axs[1].set_anchor('NW')
+
+# %%
+# For more complicated situations it may be necessary to place the axes
+# manually. In the example above, even approximating figure aspect ratio, there
+# are still blank spaces (that can be trimmed in a final product by
+# ``bbox_inches="tight"`` in `~.Figure.savefig`). The procedure is also not
+# very general.  For instance, if the axes had been arranged vertically instead
+# of horizontally, setting the height aspect ratio would not have helped
+# because the axes are wider than they are tall.
+#
+# Manual placement
+# ================
+#
+# We can manually place axes when they are created by passing a position to
+# `~.Figure.add_axes`.  This position takes the form ``[left bottom width
+# height]`` and is in units that are a fraction of the figure width and height.
+# Here we decide how large to make the axes based on the size of the images,
+# and add a small buffer of 0.35 inches.  We do all this at 100 dpi.
```

Comment:
```suggestion
# Explicit placement
# ==================
# The above approach with adjusting ``figsize`` and ``width_ratios`` does
# not generalize well, because it needs manual parameter tuning, and
# possibly even code changes to using ``height_ratios`` instead of 
# ``width_ratios`` depending on the aspects and layout of the images.
#
# We can alternative calculate positions explicitly and place Axes at absolute
# coordinates using `~.Figure.add_axes`. This takes the position in the form
# ``[left bottom width height]`` and is in
# :ref:`figure coordinates <transforms_tutorial>`. In the following, we
# determine figure size and Axes positions so that one image data point
# is rendered exactly to one figure pixel.
```